Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -2443,6 +2445,24 @@ FloatRect RenderElement::referenceBoxRect(CSSBoxType boxType) const
         return referenceBox;
     };
 
+    auto determineSVGViewport = [&]() {
+        const auto* viewportElement = downcast<SVGElement>(element());
+
+#if ENABLE(LAYER_BASED_SVG_ENGINE)
+        // RenderSVGViewportContainer is the only possible anonymous renderer in the SVG tree.
+        if (!viewportElement && document().settings().layerBasedSVGEngineEnabled()) {
+            ASSERT(is<RenderSVGViewportContainer>(this));
+            ASSERT(isAnonymous());
+            viewportElement = &downcast<RenderSVGViewportContainer>(*this).svgSVGElement();
+        }
+#endif
+
+        // FIXME: [LBSE] Upstream: Cache the immutable SVGLengthContext per SVGElement, to avoid the repeated RenderSVGRoot size queries in determineViewport().
+        ASSERT(viewportElement);
```

Comment:
We are making things more strict for non LBSE, are we sure the ASSERT will not be hit?

Reply:
We don't to my best knowledge.
Previously the first check in the function was " if (!is<SVGElement>(element()))" which turned into " if (element && !is<SVGElement>(element()))". For an anonymous element we returned early before, now we continue processing. However there are no anonymous elements in the SVG DOM tree that could create renderers in the legacy engine. Same for LBSE with the only exception is that we manually create an anonymous renderer to wrap the outermost <svg> children...

Debug bots should catch this too, no?

Reply:
Right, I guess this is okay.